src/DotNetLightning.Core/Channel/CommitmentsModule.fs

```diff
@@ -437,6 +438,106 @@ module RemoteForceClose =
         return obscuredCommitmentNumber
     }
 
+    let createPunishmentTx (perCommitmentSecret: PerCommitmentSecret)
+                           (commitments: Commitments)
+                           (localChannelPrivKeys: ChannelPrivKeys)
+                           (network: Network) 
+                               : TransactionBuilder =
+
+        let localChannelPubKeys = commitments.LocalParams.ChannelPubKeys
+        let remoteChannelPubKeys = commitments.RemoteParams.ChannelPubKeys
+
+        let perCommitmentPoint = perCommitmentSecret.PerCommitmentPoint()
+
+        let localCommitmentPubKeys =
+            perCommitmentPoint.DeriveCommitmentPubKeys localChannelPubKeys
+
+        let remoteCommitmentPubKeys =
+            perCommitmentPoint.DeriveCommitmentPubKeys remoteChannelPubKeys
+
+        let transactionBuilder = network.CreateTransactionBuilder()
+
+        let toRemoteScriptPubKey =
+            localCommitmentPubKeys
+                .PaymentPubKey
+                .RawPubKey()
+                .WitHash.ScriptPubKey
+
+        let toLocalScriptPubKey =
+            Scripts.toLocalDelayed
+                localCommitmentPubKeys.RevocationPubKey
+                commitments.RemoteParams.ToSelfDelay
+                remoteCommitmentPubKeys.DelayedPaymentPubKey
+
+        let toLocalWitScriptPubKey = toLocalScriptPubKey.WitHash.ScriptPubKey
+
+        let commitFee = commitTxFee 
+                            commitments.RemoteParams.DustLimitSatoshis 
+                            commitments.RemoteCommit.Spec
+
+        let (toLocalAmount, toRemoteAmount) =
+            if (commitments.LocalParams.IsFunder) then
+                (commitments.RemoteCommit.Spec.ToLocal.Satoshi
+                 |> Money.Satoshis),
+                (commitments.RemoteCommit.Spec.ToRemote.Satoshi
+                 |> Money.Satoshis) - commitFee
+            else
+                (commitments.RemoteCommit.Spec.ToLocal.Satoshi
+                 |> Money.Satoshis) - commitFee,
+                (commitments.RemoteCommit.Spec.ToRemote.Satoshi
+                 |> Money.Satoshis)
+
+        let outputs = 
+            seq {
+                if toLocalAmount > commitments.RemoteParams.DustLimitSatoshis then
+                    yield TxOut(toLocalAmount, toLocalWitScriptPubKey)
+
+                if toRemoteAmount > commitments.LocalParams.DustLimitSatoshis then
+                    yield TxOut(toRemoteAmount, toRemoteScriptPubKey)
+            }
+            |> List.ofSeq
+            |> sortNonHtlcTxOut
```

Comment:
Rather than adding another new function there's already a `TxOut.LexicographicCompare` method which is what's used by `makeClosingTx` to do this.